### DISCUSSION TITLE
Build failed in v3.3.4 on Mac Silicon using asdf

### DISCUSSION THREAD

**Discussion by lulunac27a** [rbenv/ruby-build#2436] [Build failures]
I tried installing Ruby version 3.3.4 on Mac silicon, but failed to build. I tried `arch -arm64`, but still get a warning:
```
/opt/homebrew/bin/rbenv-install: line 164: rbenv-hooks: command not found
dyld[90159]: Library not loaded: /opt/local/lib/libintl.8.dylib
  Referenced from: <C61A059B-DA84-3102-B424-C038FD98043B> /opt/local/bin/greadlink
  Reason: tried: '/opt/local/lib/libintl.8.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64e' or 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/opt/local/lib/libintl.8.dylib' (no such file), '/opt/local/lib/libintl.8.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64e' or 'arm64')), '/usr/local/lib/libintl.8.dylib' (no such file), '/usr/lib/libintl.8.dylib' (no such file, not in dyld cache)
```
I even use `asdf install ruby 3.3.4 --architecture=arm64`, but I still get the same warning above, what can I do?

**Comment by mislav** [comment]
Hi, it looks like the `greadlink` binary installed by Homebrew on your system is busted. rbenv will optionally use the `greadlink` tool if it finds it in PATH, but is not necessary for normal operation of rbenv. However, executing `greadlink` on your system fails due to a dynamic library that failed to load; presumably due to the fact that it was built for amd64 architecture (and you're running arm64).

The `greadlink` tool is provided by the "coreutils" Homebrew formula. You could try uninstalling it: `brew uninstall coreutils`

General protip: you can check which Homebrew formula provides a binary by looking at where the binary symlink resolves to:
```
$ ls -l `which greadlink`
lrwxr-xr-x  1 mislav  admin  37 Aug 27 14:26 /opt/homebrew/bin/greadlink -> ../Cellar/coreutils/9.5/bin/greadlink
```
The first directory under "Cellar" is the name of the Homebrew formula.

Alternatively, if something depends on "coreutils" on your system, you could reinstall it instead of uninstalling it: `brew reinstall coreutils`. That should recompile it for the correct architecture.

**Comment by lulunac27a** [comment]
Still get the same error while running `greadlink` after I use `brew reinstall coreutils`:
```
dyld[29663]: Library not loaded: /opt/local/lib/libintl.8.dylib
  Referenced from: <C61A059B-DA84-3102-B424-C038FD98043B> /opt/local/bin/greadlink
  Reason: tried: '/opt/local/lib/libintl.8.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64e' or 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/opt/local/lib/libintl.8.dylib' (no such file), '/opt/local/lib/libintl.8.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64e' or 'arm64')), '/usr/local/lib/libintl.8.dylib' (no such file), '/usr/lib/libintl.8.dylib' (no such file, not in dyld cache)
[1]    29663 abort      greadlink
```

**Reply by mislav** [reply]
Puzzling. I do have "coreutils" but I do not have the `/opt/local` directory on my machine, so I do not know what `/opt/local/lib/libintl.8.dylib` is.

1. What is your output of `otool -L /opt/homebrew/bin/greadlink`?
2. Does `/opt/local` appear at all in your environment variables? `env | grep opt/local`
3. Did you ever use MacPorts to install software on this machine in addition to Homebrew?

**Reply by mislav** [reply]
⚠️ *Potentially dangerous*: if this was an artifact from an outdated setup that's no longer necessary, you could try deleting `/opt/local` entirely. My hunch is that this directory is not needed for anything Homebrew-related at all, since Homebrew keeps everything under `/opt/homebrew`. Furthermore, macOS does not keep anything under `/opt/local` to the best of my knowledge. However, removing this directory might break other software if they ever linked to it.

**Reply by monfresh** [reply]
`/opt/local` is used by MacPorts. My guess is that @lulunac27a  either has MacPorts installed now, or used to, and they didn't properly uninstall it. Homebrew and MacPorts are not compatible with one another.

**Reply by monfresh** [reply]
And as part of properly uninstalling MacPorts, it also means removing any references to it or the `/opt/local` folder from your shell startup file (such as ~/.zshrc)

**Reply by lulunac27a** [reply]
1. 
```
/opt/homebrew/bin/greadlink:
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1345.100.2)
```
2. `/opt/local/bin:/opt/local/sbin:/opt/local/share/man`
